tests/unit_tests/test_view_model.py

```diff
@@ -0,0 +1,119 @@
+
+
+def test_done_items_returns_empty_list_if_all_items_are_to_do():
+    items = [to_do_item(1), to_do_item(2)]
+    view_model = ViewModel(items)
+    assert view_model.done_items == []
+
+
+def test_should_show_all_done_items_if_there_are_less_than_five_done_items():
+    items = [done_item(i) for i in range(3)]
+    view_model = ViewModel(items)
+    assert view_model.should_show_all_done_items
+
+
+def test_should_not_show_all_done_items_if_there_are_five_done_items():
+    items = [done_item(i) for i in range(5)]
+    view_model = ViewModel(items)
+    assert not view_model.should_show_all_done_items
+
+
+def test_should_not_show_all_done_items_if_there_are_more_than_five_done_items():
+    items = [done_item(i) for i in range(10)]
+    view_model = ViewModel(items)
+    assert not view_model.should_show_all_done_items
+
+def test_does_not_count_todo_items_towards_the_five_done_items_limit():
+    items = [done_item(i) for i in range(4)] + [to_do_item(i + 4) for i in range(10)]
+    view_model = ViewModel(items)
+    assert view_model.should_show_all_done_items
+
+def test_recent_done_items_includes_items_last_modified_today():
+    recent_item = done_item(1, datetime.now())
+    items = [recent_item]
+    view_model = ViewModel(items)
+    assert recent_item in view_model.recent_done_items
+
+
+def test_recent_done_items_excludes_items_last_modified_yesterday():
+    yesterday_item = done_item(1, datetime.now() - timedelta(days=1))
+    items = [yesterday_item]
+    view_model = ViewModel(items)
+    assert not yesterday_item in view_model.recent_done_items
+
+
+def test_recent_done_items_excludes_items_last_modified_late_yesterday():
+    now = datetime.now()
+    hours_since_10_pm_yesterday = 2 + now.hour
+    late_yesterday = now - timedelta(days=1, hours=hours_since_10_pm_yesterday)
+    late_yesterday_item = done_item(1, late_yesterday)
+    items = [late_yesterday_item]
+    view_model = ViewModel(items)
+    assert not late_yesterday_item in view_model.recent_done_items
+
+
+def test_older_done_items_excludes_items_last_modified_today():
+    recent_item = done_item(1, datetime.now())
+    items = [recent_item]
+    view_model = ViewModel(items)
+    assert not recent_item in view_model.older_done_items
+
+
+def test_recent_done_items_excludes_items_last_modified_yesterday():
+    yesterday_item = done_item(1, datetime.now() - timedelta(days=1))
+    items = [yesterday_item]
+    view_model = ViewModel(items)
+    assert yesterday_item in view_model.older_done_items
+
+
+def test_recent_done_items_excludes_items_last_modified_late_yesterday():
+    now = datetime.now()
+    hours_since_10_pm_yesterday = 2 + now.hour
+    late_yesterday = now - timedelta(days=1, hours=hours_since_10_pm_yesterday)
+    late_yesterday_item = done_item(1, late_yesterday)
+    items = [late_yesterday_item]
+    view_model = ViewModel(items)
+    assert late_yesterday_item in view_model.older_done_items
+
+
+def done_item(index, last_modified = None):

```

Comment:
This is really minor and I wouldn't normally mention it if everything else wasn't close to perfect, but I'd prefer to put `done_item` and `to_do_item` at the top of the file, since when I was reading these tests I was a bit confused about what they were